cliquet/tests/resource/test_views.py

```diff
@@ -638,6 +638,51 @@ def test_cache_control_provides_no_cache_by_default(self):
         self.assertNotIn('Expires', resp.headers)
 
 
+class VaryHeaderTest(BaseWebTest, unittest.TestCase):
+    collection_url = '/toadstools'
+
+    def setUp(self):
+        super(VaryHeaderTest, self).setUp()
+
+        body = {'data': MINIMALIST_RECORD}
+        resp = self.app.post_json(self.collection_url, body,
+                                  headers=self.headers)
+        self.record = resp.json['data']
+        self.record_url = self.get_item_url()
+
+    def get_app_settings(self, extras=None):
+        settings = super(VaryHeaderTest, self).get_app_settings(extras)
+        settings['toadstool_read_principals'] = 'system.Everyone'
+        return settings
+
+    def test_vary_header_is_not_set_if_anonymous_on_hello_page(self):
```

Comment:
Well actually it's not how it's supposed to work. The answer will vary, depending the authentication you put. If you don't add the Vary when you're anonymous, then the anonymous version will be cached and served even when you're authenticated.